plugins/experimental/stale_while_revalidate/stale_while_revalidate.c

```diff
@@ -184,39 +134,38 @@ free_request_state(StateInfo *state)
     void *ptr;
 
     TSMutexLock(state->plugin_config->troot_mutex);
-    ptr = tfind(state->req_info->effective_url, &(state->plugin_config->troot), xstrcmp);
+    ptr = tfind(state->req_info->effective_url, &(state->plugin_config->troot), XSTRCMP);
```

Comment:
Why change the `strcmp` usage to save a `jmp` but then add an extra level of pointer dereferencing with `req_info`?

Reply:
Firstly .. it's not a "jmp" ... but a simplification of another code-line-maint that one doesn't mistrust, (as well as a second O(n) cache / pipeline hit, of course).  Cache is the big deal these days.

Someone can alter or have strange linkage for _xstrcmp but no one will stuff anything so surprising into a simple #define of the same function pointer.   It's unable to be corrupted, unlike a trampoline-inline-function that's hidden away somewhere.

For the req_info->?   Because the number-of-changed-characters is very very very very very important to some.  There are tons of places where malloc(8) and malloc(16) and malloc(10) are being called.   The system is designed around tons of layers of indirection.... and that has to stay the same in order to show the code is working the same.

Const would be better ... giving a hint to the compiler that it never gets another value .. but that's too much for this little segment.   Just far far better to get rid of endlessly overcomplex malloc/free pairs that aren't needed.

http://en.cppreference.com/w/cpp/language/raii